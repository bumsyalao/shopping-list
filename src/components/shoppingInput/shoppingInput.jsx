import React, { Component } from 'react';
import { connect } from 'react-redux';
import './shoppingInput.css';
import ShoppingCart from '../shoppingCart/shoppingCart';
import { setBudget, addItem, getItems } from '../../actions/shoppingListActions';


export class ShoppingInput extends Component {

  constructor(props) {
    super(props);

    this.state = {
      budget: '',
      product: '',
      quantity: null,
      price: null,
    }

    this.onChange = this.onChange.bind(this);
    this.onSetBudget = this.onSetBudget.bind(this);
    this.onAddItem = this.onAddItem.bind(this);

  }

  componentDidMount() {
    if(this.props.items) {
    this.props.getItems();
  }}
  onChange(event) {
    this.setState({
      [event.target.id]: event.target.value
    })
  }

  onSetBudget(event) {
    event.preventDefault();
    this.props.setBudget(this.state.budget);

  }

  onAddItem(event){
    event.preventDefault();

    const Items = {
      product: this.state.product,
      quantity: this.state.quantity,
      price: this.state.price,
    }

    this.props.addItem(Items);

  }

  renderBudget(budget) {
    return(
      <div>
        <p>{`Your Budget is ${budget}`}</p>
      </div>
    )
  }

  renderShoppingCart(shoppingItems) {
    return(
      <ShoppingCart items={shoppingItems}/>
    )
  }

  render() {
    return (
      <div>
        <div className="budget-input">
          <input
            id="budget"
            value={this.state.budget}
            onChange={this.onChange}
            type="text"
            placeholder="what's your budget?" />
          <button
            id="set-budget"
            disabled={!this.state.budget || this.props.budget}
            onClick={this.onSetBudget}>
          SET BUDGET</button>
        </div>
        <div id="input-form" className="shop-input">
          <input
            id="product"
            value={this.state.product}
            onChange={this.onChange}
            type="text"
            maxlength="30"
            placeholder="what you buying?" />
          <input
            id="quantity"
            value={this.state.quantity}
            onChange={this.onChange}
            type="text"
            placeholder="quantity" />
          <input
            id="price"
            value={this.state.price}
            onChange={this.onChange}
            type="text"
            placeholder="price" />
          <button
            onClick={this.onAddItem}>
          ADD</button>
        </div>
        <div>
          {this.renderBudget(this.props.budget || 0)}
        </div>
        <div>
          {this.renderShoppingCart(this.props.Items)}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  ...state,
  budget: state.shoppingListReducer.budget,
  Items: state.shoppingListReducer.items,
});



export default connect(mapStateToProps, { setBudget, addItem, getItems })(ShoppingInput);
