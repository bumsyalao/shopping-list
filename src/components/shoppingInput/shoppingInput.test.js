import React from 'react';
import { shallow, configure } from 'enzyme';
import adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'

import { ShoppingInput } from './shoppingInput';

configure({ adapter: new adapter() });
const spy = jest.fn();

describe('ShoppingInput component', () => {
  const props = {
    budget: 5000,
    items: {
      product: 'shoes',
      price: 3,
      quantity: 300,
    },
  
    getItems: jest.fn(() => Promise.resolve()),
    setBudget: jest.fn(() => Promise.resolve()),
    addItem: jest.fn(() => Promise.resolve()),
  };
  const component = shallow(
    <ShoppingInput {...props} />
  );

  it('should render component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should set state when onChange function is called', () => {
    component.instance().onChange({target: {value: 500, id: 'budget'}});
    expect(component.state('budget')).toEqual(500)
  });

  it('should set budget when set budget is clicked', () => {
    component.instance().onChange({target: {value: 500, id: 'budget'}});
    expect(component.find('#set-budget').length).toEqual(1);
    expect(component.state('budget')).toEqual(500);
  })
});

