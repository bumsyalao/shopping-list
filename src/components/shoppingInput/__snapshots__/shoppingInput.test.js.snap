// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingInput component should render component 1`] = `ShallowWrapper {}`;
